### PR TITLE
Refactor doc of NNI model compression

### PR DESCRIPTION
**Pull request by QuanluZhang**

### CODE REVIEW

File: docs/en_US/Compressor/Framework.md [hunk 1]
```diff
@@ -52,7 +58,7 @@ A `pruner` is responsible for:
 3. Use `weight masker` to calculate masks of layers while pruning.
 4. Export pruned model weights and masks.
 
-## Implement a new pruning algorithm
+### Implement a new pruning algorithm
 
 Implementing a new pruning algorithm requires implementing a `weight masker` class which shoud be a subclass of `WeightMasker`, and a `pruner` class, which should a subclass `Pruner`.
```
Comment (colorjam) [inline docs/en_US/Compressor/Framework.md:63]: should a subclass -> should be a subclass of

File: docs/en_US/Compressor/Framework.md [hunk 2]
```diff
@@ -142,3 +148,125 @@ self.pruner.remove_activation_collector(collector_id)
```
Comment (colorjam) [inline docs/en_US/Compressor/Framework.md:148]: A basic pruner -> A basic`pruner`

File: docs/en_US/Compressor/Pruner.md [hunk 3]
```diff
@@ -1,5 +1,4 @@
-Pruner on NNI Compressor
-===
+# Supported Pruning Algorithms on NNI
 
 Index of supported pruning algorithms
```
Comment (colorjam) [inline docs/en_US/Compressor/Pruner.md:3]: Suggest to organize with fine-grained pruning / structural pruning

File: docs/en_US/Compressor/QuickStart.md [hunk 4]
```diff
@@ -1,8 +1,12 @@
-# Quick Start to Compress a Model
+# Tutorial for Model Compression
+
+In this tutorial, we use the [first section](#quick-start-to-compress-a-model) to quickly go through the usage of model compression on NNI. Then use the [second section](#detailed-usage-guide) to explain more details of the usage.
+
+## Quick Start to Compress a Model
 
 NNI provides very simple APIs for compressing a model. The compression includes pruning algorithms and quantization algorithms. The usage of them are the same, thus, here we use slim pruner as an example to show the usage.
```
Comment (colorjam) [inline docs/en_US/Compressor/QuickStart.md:8]: suggest adding a link for [slim pruner](https://github.com/microsoft/nni/blob/master/docs/en_US/Compressor/Pruner.md#slim-pruner)

File: docs/en_US/Compressor/Framework.md (hunk 1 above)
Comment (QuanluZhang) [reply docs/en_US/Compressor/Framework.md:67]: good catch, thanks!

File: docs/en_US/Compressor/Framework.md (hunk 2 above)
Comment (QuanluZhang) [reply docs/en_US/Compressor/Framework.md:152]: which one?

File: docs/en_US/Compressor/Pruner.md (hunk 3 above)
Comment (QuanluZhang) [reply docs/en_US/Compressor/Pruner.md:3]: could you give me a version of such organization?

File: docs/en_US/Compressor/QuickStart.md (hunk 4 above)
Comment (QuanluZhang) [reply docs/en_US/Compressor/QuickStart.md:12]: good point, thanks!

File: docs/en_US/Compressor/Framework.md (hunk 2 above)
Comment (colorjam) [reply docs/en_US/Compressor/Framework.md:152]: It's in the old version, at Line83 `A basic pruner looks likes this:`

File: docs/en_US/Compressor/Pruner.md (hunk 3 above)
Comment (colorjam) [reply docs/en_US/Compressor/Pruner.md:3]: Sure~ let me try it.

File: docs/en_US/Compressor/Framework.md [hunk 5]
```diff
@@ -1,6 +1,16 @@
-# Design Doc
+# Customize A New Compression Algorithm
```
Comment (scarlett2018) [inline docs/en_US/Compressor/Framework.md:2]: From the Overview.md
> Details about how model compression framework works can be found in here.

this doc seems to explain "how model compression framework works", right? But now, it seems turn into a tutorial??

if we'd like to explain how the framework works, we might want to add a component diagram or workflow diagram to ease the reading..

File: docs/en_US/Compressor/Framework.md [hunk 6]
```diff
@@ -1,6 +1,16 @@
-# Design Doc
```
Comment (scarlett2018) [inline docs/en_US/Compressor/Framework.md:1]: From the Overview.md
> Details about how model compression framework works can be found in here.

this doc seems to explain "how model compression framework works", right? But now, it seems turn into a tutorial??

if we'd like to explain how the framework works, we might want to add a component diagram or workflow diagram to ease the reading..

File: docs/en_US/Compressor/Pruner.md [hunk 7]
```diff
@@ -181,6 +188,18 @@ pruner.compress()
...
+| Model         | Error(paper/ours) | Parameters | Pruned    |
+| ------------- | ----------------- | ---------- | --------- |
+| VGGNet        | 6.34/6.40     | 20.04M   |           |
+| Pruned-VGGNet | 6.20/6.26     | 2.03M    | 88.5% |
+
+The experiments code can be found at [examples/model_compress]( https://github.com/microsoft/nni/tree/master/examples/model_compress/)
+
+***
 
 ## WeightRankFilterPruner
```
Comment (chicm-ms) [inline docs/en_US/Compressor/Pruner.md:204]: Would you remove the ` WeightRankFilterPruner`?  It has been removed in code.

File: docs/en_US/Compressor/Pruner.md [hunk 8]
```diff
@@ -355,6 +387,7 @@ You can view example for more information
 - **sparsity:** How much percentage of convolutional filters are to be pruned.
 - **op_types:** Only Conv2d is supported in ActivationMeanRankFilterPruner.
 
+***
 
 ## GradientRankFilterPruner
```
Comment (chicm-ms) [inline docs/en_US/Compressor/Pruner.md:392]: `GradientRankFilterPruner` can also be removed.

File: docs/en_US/Compressor/Pruner.md [hunk 9]
```diff
@@ -292,6 +322,8 @@ pruner.compress()
 - **sparsity:** This is to specify the sparsity operations to be compressed to
 - **op_types:** Only Conv1d and Conv2d is supported in L2Filter Pruner
 
+***
+
 ## ActivationRankFilterPruner
```
Comment (chicm-ms) [inline docs/en_US/Compressor/Pruner.md:327]: `ActivationRankFilterPruner` also needs to be removed.

File: docs/en_US/Compressor/Pruner.md (hunk 3 above)
Comment (chicm-ms) [reply docs/en_US/Compressor/Pruner.md:3]: The `WeightRankFilterPruner`, `ActivationRankFilterPruner` and `GradientRankFilterPruner` have been removed from code. Suggest to flatten the structured pruners.

Comment (chicm-ms) [reply docs/en_US/Compressor/Pruner.md:3]: And suggest move lottery ticket to another category , or same category as `AGP`

File: docs/en_US/Compressor/Framework.md [hunk 10]
```diff
@@ -1,6 +1,16 @@
...
+.. contents::
+```
+
+To simplify writing a new compression algorithm, we design programming interfaces which are simple but flexible enough. There are interfaces for pruning and quantization respectively. Below, we first demonstrate how to customize a new pruning algorithm and then demonstrate how to customize a new quantization algorithm.
+
+## Customize a new pruning algorithm
+
+To better demonstrate how to customize a new pruning algorithm, it is necessary for users to first understand the framework for supporting various pruning algorithms in NNI.
+
+### Framework overview for pruning algorithms
```
Comment (squirrelsc) [inline docs/en_US/Compressor/Framework.md:13]: The title doesn't help to understand purpose of its content. The content should like a step by step guidance to create user's puring algorithm.

File: docs/en_US/Compressor/Framework.md (hunk 6 above)
Comment (QuanluZhang) [reply docs/en_US/Compressor/Framework.md:1]: @chicm-ms will update this part (add a diagram) in a following pr after this pr is merged

File: docs/en_US/Compressor/Framework.md (hunk 10 above)
Comment (QuanluZhang) [reply docs/en_US/Compressor/Framework.md:13]: also leave it to @chicm-ms :)

File: docs/en_US/Compressor/Framework.md (hunk 2 above)
Comment (QuanluZhang) [reply docs/en_US/Compressor/Framework.md:152]: fixed

File: docs/en_US/Compressor/Pruner.md (hunk 3 above)
Comment (QuanluZhang) [reply docs/en_US/Compressor/Pruner.md:3]: updated, could @chicm-ms make further changes in your pr? thanks!

**Comment by chicm-ms** [comment]
@scarlett2018  @squirrelsc @colorjam   I will create a  new PR to address your comments.

**Pull request merged**